src/internal/rest.rs

```diff
@@ -185,9 +184,9 @@ impl<'a> Authorization<'a> {
 pub struct SignatureUrlString(String);
 
 impl SignatureUrlString {
-    pub fn new(encoded_full_url: &str) -> Result<SignatureUrlString, reqwest::UrlError> {
+    pub fn new(encoded_full_url: &str) -> Result<SignatureUrlString, url::ParseError> {
         let parsed_url = Url::parse(encoded_full_url)?;
-        let query_str_format = |q: &str| format!("?{}", q);
+        let query_str_format = |q: &str| format!("?{}", q.replace("%27", "'"));
```

Comment:
There should probably be a comment here explaining what's going on here.

Reply:
Yep.